Tests/UnitTests/Core/TrackFitting/KalmanFitterTests.cpp

```diff
@@ -150,482 +65,124 @@ Acts::CurvilinearTrackParameters makeParameters() {
                                           1_e, cov);
 }
 
-const GeometryContext geoCtx;
-const MagneticFieldContext magCtx;
-const CalibrationContext calCtx;
-
-// detector geometry
-CubicTrackingGeometry geometryStore(geoCtx);
-const auto geometry = geometryStore();
-// expected number of measurements for the given detector
-constexpr size_t nMeasurements = 6u;
-
-// detector resolutions
-const MeasurementResolution resPixel = {MeasurementType::eLoc01,
-                                        {25_um, 50_um}};
-const MeasurementResolution resStrip0 = {MeasurementType::eLoc0, {100_um}};
-const MeasurementResolution resStrip1 = {MeasurementType::eLoc1, {150_um}};
-const MeasurementResolutionMap resolutions = {
-    {GeometryIdentifier().setVolume(2), resPixel},
-    {GeometryIdentifier().setVolume(3).setLayer(2), resStrip0},
-    {GeometryIdentifier().setVolume(3).setLayer(4), resStrip1},
-    {GeometryIdentifier().setVolume(3).setLayer(6), resStrip0},
-    {GeometryIdentifier().setVolume(3).setLayer(8), resStrip1},
-};
-
-// simulation propagator
-const auto simPropagator = makeStraightPropagator(geometry);
+// Instantiate the tester
+const FitterTester tester;
 
 // reconstruction propagator and fitter
 const auto kfLogger = getDefaultLogger("KalmanFilter", Logging::INFO);
-const auto kfZeroPropagator = makeConstantFieldPropagator(geometry, 0_T);
+const auto kfZeroPropagator =
+    makeConstantFieldPropagator<ConstantFieldStepper>(tester.geometry, 0_T);
 const auto kfZero = KalmanFitter(kfZeroPropagator);
 
 std::default_random_engine rng(42);
 
+auto makeDefaultKalmanFitterOptions() {
+  return KalmanFitterOptions(tester.geoCtx, tester.magCtx, tester.calCtx,
+                             getExtensions(), LoggerWrapper{*kfLogger},
```

Comment:
I guess if there's this function, there's no need for a separate `getExtensions` anymore is there? You consider manually inlining that here.